src/index.ts

```diff
@@ -108,7 +108,14 @@ const addOAuthInterceptor = (
 
     const paramsToSign = { ...oauthParams };
     if (config.params) {
-      for (const [k, v] of Object.entries(config.params)) {
+      let entries: [string, string][]
+      if (config.params instanceof URLSearchParams) {
+        entries = []
+        config.params.forEach((value, key) => entries.push([key, value]))
+      } else {
+        entries = Object.entries(config.params)
+      }
+      for (const [k, v] of entries) {
         paramsToSign[k] = String(v);
```

Comment:
I suppose technically a param could be added twice in the `URLSearchParams` and this would only put it into the output once.

I wonder if something like this would work:

```suggestion
      for (const [k, v] of entries) {
        const oldValue = paramsToSign[k];
        if(!oldValue) {
          paramsToSign[k] = String(v);
        } else if(Array.isArray(oldValue)) {
          oldValue.push(String(v));
        } else {
          paramsToSign[k] = [oldValue, String(v)];
        }
```

Do you happen to have time to test this idea out?  That'll avoid a bug report later about having multiple values for a parameter not working when using URLSearchParams.